runtime/jcl/common/mgmtthread.c

```diff
@@ -1114,35 +1103,26 @@ handlerMonitorWaited(J9HookInterface** hook, UDATA eventNum, void* eventData, vo
  * These classes should be loaded only if ThreadMXBean is used.
  * @pre must not have VM access
  * @todo When should these references be deleted?
- * @param[in] env
- * @return error status
- * @retval JNI_OK success
- * @retval JNI_ENOMEM OutOfMemoryError is set
- * @retval JNI_ERR FindClass() or GetMethodID() failed, an exception is set
+ * @param[in] env The JNI env
+ * @param[in] jcls The Class (static method)
  */
-static jint
-initIDCache(JNIEnv *env)
+void JNICALL
+Java_openj9_management_internal_IDCacheInitializer_initIDCache(JNIEnv *env, jclass jcls)
 {
-	jclass oom;
-	jclass cls;
-	jclass gcls;
-	jmethodID mid;
+	jclass oom = NULL;
+	jclass cls = NULL;
+	jclass gcls = NULL;
+	jmethodID mid = NULL;
 	jint err = JNI_OK;
```

Comment:
Rather than keeping this local, we should check if an exception is pending in the failure case and throw OOM if there is not.

Reply:
done, removed the local and made the check.